aopy/data/bmi3d.py

```diff
@@ -1004,7 +1117,176 @@ def get_ts_data_segment(preproc_dir, subject, te_id, date, start_time, end_time,
 
     return data, samplerate
 
+@lru_cache(maxsize=1)
+def extract_lfp_features(preproc_dir, subject, te_id, date, decoder, samplerate=None, channels=None,
```

Comment:
It isn't clear that the decoding is happening in this function based on the name. I think you should either just rename to something like "tabulate_lfp_features_and_decode" or pull the decoding out of this function and make a wrapper.

Reply:
Actually, since this returns either features or states I think they should be separate functions.